diff --git a/prosynth/souffle/configure.ac b/prosynth/souffle/configure.ac
index 4fc7bfd..a540b7a 100644
--- a/prosynth/souffle/configure.ac
+++ b/prosynth/souffle/configure.ac
@@ -25,7 +25,6 @@ AM_INIT_AUTOMAKE([foreign subdir-objects 1.9 tar-pax])
 AM_MAINTAINER_MODE
 AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_MACRO_DIRS([m4])
-AC_CONFIG_MACRO_DIR([m4])
 
 dnl Do not set default CFLAGS and CXXFLAGS
 ENV_CXXFLAGS="$CXXFLAGS"
